e2e/imgbuild/imgbuild.go

```diff
@@ -880,12 +832,36 @@ func (c *imgBuildTests) buildEncryptPassphrase(t *testing.T) {
 
 	// First with the command line argument
 	passphraseInput := []e2e.SingularityConsoleOp{
-		e2e.ConsoleSendLine(passphrase),
+		e2e.ConsoleSendLine(e2e.Passphrase),
 	}
 	imgPath1 := filepath.Join(tempDir, "encrypted_cmdline_option.sif")
-	cmdArgs := []string{"--encrypt", "--passphrase", imgPath1, "library://alpine:latest"}
+	cmdArgs := []string{"--passphrase", imgPath1, "library://alpine:latest"}
+	c.env.RunSingularity(
+		t,
+		e2e.AsSubtest("passphrase flag"),
+		e2e.WithCommand("build"),
+		e2e.WithPrivileges(true),
+		e2e.WithArgs(cmdArgs...),
+		e2e.ConsoleRun(passphraseInput...),
+		e2e.ExpectExit(
+			expectedExitCode,
+			e2e.ExpectError(e2e.ContainMatch, expectedStderr),
+		),
+	)
+	// If the command was supposed to succeed, we check the image
+	if expectedExitCode == 0 {
+		c.ensureImageIsEncrypted(t, imgPath1)
+	}
+	err = os.Remove(imgPath1)
```

Comment:
I think the CI failure can be fixed by only removing the image if the exit code is 0, or having it build to a directory that is removed automatically when the test ends. Right now if it's supposed to fail because of an incorrect cryptsetup version it will try to remove the non-existent resulting image

Reply:
Yes. I changed the code to instead let the cleanup function delete everything and use a different path for the image for each case. I hope it will make the code easier to read.